packages/website-backend/src/services/real-time/MutationEventSender.ts

```diff
@@ -0,0 +1,34 @@
+import { MutantResult } from 'mutation-testing-report-schema';
+import { ServerResponse } from 'http';
+
+export class MutationEventSender {
+  #response: ServerResponse;
+
+  constructor(res: ServerResponse, cors: string) {
+    this.#response = res;
+    this.#response.writeHead(200, {
+      'Content-Type': 'text/event-stream',
+      'Cache-Control': 'no-cache',
+      Connection: 'keep-alive',
+      'Access-Control-Allow-Origin': cors,
+    });
+    this.#response.on('close', () => this.#destroy());
```

Comment:
I assume this `close` event is raised when the client disconnects. If so, shouldn't the `response` also be removed from the `MutationEventResponseHandler`?

Reply:
Maybe, right now I remove the `MutationEventResponseHandler` when real-time testing is done:
https://github.com/stryker-mutator/stryker-dashboard/blob/41a529c3713ca473b0db25ae3fbb03f2ac613272/packages/website-backend/src/api/RealTimeReportsController.ts#L116

But I guess it does make sense if a user disconnects, that nothing is sent to that response any more.

Reply:
It now emits an event :)